feat: interfaces and structures for feed, which support query table-like query operations

vaaandark opened the pull request:
- [x] Basic design of data structure.
- [x] Serialization and persistence of data structures
- [x] Combine with articles.

vaaandark committed:
feat: basic structures for feed

vaaandark committed:
feat: basic interfaces for feed

vaaandark committed:
doc: add doc comments for module feed and opml

jyi2ya commented on src/feed.rs:
```diff
@@ -0,0 +1,538 @@
+    /// Set entry belongs to a folder.
+    pub fn set_belonging(&mut self, belong_to: &FolderUuid) {
+        self.belong_to = Some(*belong_to);
+    }
```
把 new_with_html_url 弄成 set_html_url(&mut self, html_url: Url) -> &mut self，set_belonging 的返回值也改成 &mut self，接着就可以欢乐链式调用了

vaaandark committed:
refactor: remove field `folders` from `Feed`

vaaandark committed:
refactor: make code more rusty

vaaandark committed:
refactor: modified the visibility of some fields, and add some getters and setters

vaaandark committed:
feat: add serialization and deserialization for feed

Use `url::Url` to replace `Reqwest::url` for serde featrue.

vaaandark committed:
Merge branch 'master' into feed

vaaandark committed:
feat: add basic support for fetching articles

There are two main problems:
- Cannot use `blocking` feature of reqwest in wasm.
- It will be blocked when fetching.

vaaandark committed:
fead: fetch articles asynchronously now

Some works has been done:
- `ehttp` to replace `reqwest`.
- modify some function returning `impl Iterator` to returning `Vec<_>`.
- add test for fetching articles.

vaaandark committed:
refactor: remove useless `RefCell` for type ArticlesMap

vaaandark committed:
feat: add field `updated` for `ArticleUuid` to sort by time
